lib/scarpe/promises.rb

```diff
@@ -284,7 +286,7 @@ def call_handlers_for(state)
             @scheduler.call(*@parents.map(&:returned_value))
           rescue => e
             if Promise.debug
```

Comment:
i dont think i can answer this but i'd say explicit module when logging as you said in meet tht shoes/scarpe is for beginners n developer poductivity? so ofc i would say something that helps ppl while debugging

Reply:
Yeah, the tradeoff is basically that if you need an explicit module, logging requires slightly more effort, and if you do it automagically then sometimes debugging does... e.g.:

~~~
    Scarpe.error(Scarpe::Promises, "error when running scheduler!")

    vs

    Scarpe.error("error when running scheduler!")  # uses caller to automatically pick up class name unless overridden
~~~

We should probably ask on the Discord, see if anybody has a strong opinion.

But it's *really* useful to be able to turn debug messages on/off per-class or per-module instead of globally.